### PR TITLE
CNV-28503: Console errors while using a template without parameters

### PR DESCRIPTION
**Pull request by avivtur**
## 📝 Description

Template without a parameter section was getting errors on the quick create, customize, and parameters tab under the template details view.

## 🎥 Demo

Before:

https://github.com/kubevirt-ui/kubevirt-plugin/assets/67270715/95101049-a682-46fa-8d4b-d69a176e8dd4

After:

https://github.com/kubevirt-ui/kubevirt-plugin/assets/67270715/b6299e88-2a63-4c18-bb28-711f620501c7

### CODE REVIEW

File: src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerCreateForm.tsx [hunk 1]
```diff
@@ -124,6 +135,65 @@ export const TemplatesCatalogDrawerCreateForm: FC<TemplatesCatalogDrawerCreateFo
...
+          });
+
+          // update context vm
+          await updateVM(
+            !isEmpty(authorizedSSHKey) ? addSecretToVM(updatedVM, authorizedSSHKey) : updatedVM,
+          );
+
+          history.push(`/k8s/ns/${namespace}/templatescatalog/review`);
+        });
+      }
```
Comment (metalice) [inline src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerCreateForm.tsx:196]: why we need it?
did i saw this function somewhere? :) 
why not just adding mandatory missing params?

Comment (avivtur) [reply src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerCreateForm.tsx:196]: This is a case where the params doesn't exist at all in a template, so I want to prepare the vm for customization.
you probably saw something similar to this, but they do differ one each other.

File: src/views/catalog/customize/utils.ts [hunk 2]
```diff
@@ -77,16 +81,12 @@ export const hasCustomizableSource = (template: V1Template): boolean => {
   return true;
 };
 
-export const extractParameterNameFromMetadataName = (template: V1Template): string => {
-  const virtualMachineObject = getTemplateVirtualMachineObject(template);
-  return virtualMachineObject?.metadata.name?.replace(/[${}"]+/g, '');
-};
+export const isNameParameterExists = (template: V1Template): boolean =>
+  !isEmpty(template?.parameters) &&
+  !isEmpty(template?.parameters?.find((param) => param?.name === NAME_INPUT_FIELD));
```
[the remark this answers is not in the document]
Comment (avivtur) [reply src/views/catalog/customize/utils.ts:86]: yes since I check if the name parameter exists or not

**Pull request merged**